html/pages/templates.inc.php

```diff
@@ -13,3 +13,76 @@
  */
 
 require_once 'includes/print-alert-templates.php';
+?>
+
+<form action="" method="post">
+    <input type="hidden" name="tool" value="convert">
+    <input type="submit" class="btn btn-primary btn-sm" value="Convert template">
+</form>
+<br />
+<?php
+if ($vars['tool'] === 'convert' || $vars['old']) {
+    if ($vars['old']) {
+        foreach (explode(PHP_EOL, $vars['old']) as $line) {
+            if (str_contains($line, '{calc')) {
+                $new .= preg_replace(
+                    [
+                        '/{calc[ ]*([\w\d\s\%\.\(\)\*\/\-\+\/]+)}/',// Replaces {calc (something*100)}
+                        '/%([\w\d]+)\.([\w\d]+)/',// Replaces %something.anything
+                    ],
+                    [
+                        '@php echo \1; @endphp ',
+                        '$value[\'string\']',
+                    ],
+                    $line
+                );
+            } else {
+                $new .= convert_template($line);
+            }
+        }
+    }
+    echo "
+    <form action='' method='post'>
+        <div class='row'>
+            <div class='col-sm-6'>
+                Paste old template here<br />
+                <textarea name='old' cols='80' rows='30'>{$vars['old']}</textarea>
+            </div>
+            <div class='col-sm-6'>
+                New<br />
+                <textarea name='old' cols='80' rows='30' disabled>$new</textarea>
+            </div>
+        </div>
+        <button type='submit' class='btn btn-success btn-sm' name='submit' value='convert'>Convert</button>
+    </form>
+    ";
+}
+
+function convert_template($old)
+{
+    $find = [
+        '/{if %([\w=\s]+)}/',// Replaces {if %something == else}
+        '/{else}/',// Replaces {else}
+        '/{\/if}/',// Replaces {/if}
+        '/{foreach %faults}/',// Replaces {foreach %faults}
+        '/{foreach %contacts}/',// Replaces {foreach %contacts}
+        '/{\/foreach}/',// Replaces {/foreach}
+        '/{calc[ ]*([\w\d\s\%\.\(\)\*\/\-\+\/]+)}/',// Replaces {calc (something*100)}
+        '/%value.string/',// Replaces %value.string
+        '/%([\w\d]+)\.([\w\d]+)/',// Replaces %something.anything
+        '/%([\w\d]+)/',// Replaces %anything
+    ];
+    $replace = [
+        '@if ($\1) ',
+        ' @else ',
+        ' @endif',
+        '@foreach ($faults as $key => $value)',
+        '@foreach ($contacts as $key => $value)',
+        '@endforeach ',
+        '@php echo \1; @endphp ',
+        '$value[\'string\']',
+        '{{ $\1[\'\2\'] }}',
+        '{{ $\1 }}',
+    ];
+    return preg_replace($find, $replace, $old);
```

Comment:
You might need this to escape dollar sign followed by number.
```php
addcslashes($old, '$')
```

Reply:
You mean from the template like `$1`? If so where would this be from as we don't have any tables or variables starting with numbers.